data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkService.java

```diff
@@ -91,59 +90,48 @@ public S3SinkService(final S3SinkConfig s3SinkConfig, final BufferFactory buffer
         numberOfRecordsSuccessCounter = pluginMetrics.counter(NUMBER_OF_RECORDS_FLUSHED_TO_S3_SUCCESS);
         numberOfRecordsFailedCounter = pluginMetrics.counter(NUMBER_OF_RECORDS_FLUSHED_TO_S3_FAILED);
         s3ObjectSizeSummary = pluginMetrics.summary(S3_OBJECTS_SIZE);
+
+        reentrantLock.lock();
```

Comment:
I don't think we need a lock here. The constructor is called once by one thread.

Reply:
Makes sense, I wasn't sure. I'll remove this with the rebase